examples/twittermap/web/public/javascripts/mapresultcache/services.js

```diff
@@ -0,0 +1,82 @@
+/*
+ * Module to cache map result data. Common services module communicates with
+ * middleware and updates resultcache module.
+ */
+'use strict';
+angular.module('cloudberry.mapresultcache', ['cloudberry.common'])
+  .service('MapResultCache', function () {
+
+    // The key-value store that stores mapResults
+    var store = new HashMap();
+    // To check if keyword in query changed
+    var currentKeywords = [""];
+    // To check if time range in query changed
+    var date = new Date();
+    date.setDate(date.getDate() - 1);
+    var currentTimeRange = {
```

Comment:
I think this var is never been used, except a reset.

Reply:
It's used in the very first request. Please let me know if it must be modified.